blocks/monitoring/aggregation.py

```diff
@@ -138,11 +138,62 @@ def get_aggregator(self):
         return aggregator
 
 
-def mean(numerator, denominator=1.):
+def mean(numerator, denominator=1., name=None):
     """Mean of quantity (numerator) over a number (denominator) values."""
     variable = numerator / denominator
     variable.tag.aggregation_scheme = Mean(numerator, denominator)
-    variable.name = numerator.name
+    if name is not None:
```

Comment:
Then maybe just add a `setattr_` utility that would return its argument? 
So that is is obvious what it does and that it mutates its argument from 
the name itself?

Dmitry Bogdanov wrote:
>
> In blocks/monitoring/aggregation.py
> <https://github.com/mila-udem/blocks/pull/780#discussion_r36044839>:
>
> > """Mean of quantity (numerator) over a number (denominator) values."""
> > variable = numerator / denominator
> > variable.tag.aggregation_scheme = Mean(numerator, denominator)
> > - variable.name = numerator.name
> > + if name is not None:
>
> Without opening a can of worms by asking ourselves whether or not
> |variable.copy()| should preserve |variable.tag|, we could just use a
> new utility function |rename| instead of |named_copy| that would
> change the name of its argument and return it. Moreover, it is a shame
> that |setattr| doesn't return its first operand, otherwise we could
> just write |setattr(mean(x, y), 'name', 'per-token-cost')|.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mila-udem/blocks/pull/780/files#r36044839>.